core/src/main/java/org/apache/iceberg/rest/RESTSerializers.java

```diff
@@ -61,10 +63,19 @@ public static void registerAll(ObjectMapper mapper) {
         .addSerializer(UnboundPartitionSpec.class, new UnboundPartitionSpecSerializer())
         .addDeserializer(UnboundPartitionSpec.class, new UnboundPartitionSpecDeserializer())
         .addSerializer(UnboundSortOrder.class, new UnboundSortOrderSerializer())
-        .addDeserializer(UnboundSortOrder.class, new UnboundSortOrderDeserializer());
+        .addDeserializer(UnboundSortOrder.class, new UnboundSortOrderDeserializer())
+        .addSerializer(TableMetadata.class, new TableMetadataSerializer());
     mapper.registerModule(module);
   }
 
+  public static class TableMetadataSerializer extends JsonSerializer<TableMetadata> {
+    @Override
+    public void serialize(TableMetadata metadata, JsonGenerator gen, SerializerProvider serializers)
+        throws IOException {
+      gen.writeRawValue(TableMetadataParser.toJson(metadata));
```

Comment:
Can't this pass the generator into `TableMetadataParser` like the others do?

Reply:
Sure. That would be fine.

Reply:
It _does_ require making a new signature `public`, but given that it matches the rest I think that's fine.